starlette/staticfiles.py

```diff
@@ -61,11 +62,11 @@ def __init__(self, scope: Scope, path: str, check_directory: str = None) -> None
             if not stat.S_ISREG(mode):
                 response = PlainTextResponse("Not Found", status_code=404)
             else:
-                send_header_only = self.scope["method"] == "HEAD"
+                request = Request(self.scope, receive)
```

Comment:
In this *particular* case don't bother to instantiate a request instance.
Just use `method = self.scope["method"]`.